apps/algorithm/search/src/test_server.py

```diff
@@ -1,2 +1,1323 @@
-def test_addition():
-    assert 1 + 2 == 3
+import os
+
+import pytest
+from elasticsearch import Elasticsearch
+from fastapi.testclient import TestClient
+
+from server import app
+
+TEST_INDEX = "test-index"
+
+client = TestClient(app)
+
+es_endpoint = os.getenv("ES_ENDPOINT")
+es = Elasticsearch([es_endpoint], verify_certs=False)
+
+
+@pytest.fixture(scope="function", autouse=True)
+def setup_and_teardown_index(monkeypatch):
+    monkeypatch.setenv("ES_INDEX", TEST_INDEX)
+
+    es.options(ignore_status=[400]).indices.create(
```

Comment:
Why are we ignoring `400` for all the tests?

Reply:
Initially I didn't want the test to fail if an index already existed, but on second thought that could lead to unpredictable state problems, so I don't want to do that.

removing that.